service/src/main/resources/db/changesets/20221118_add_create_date_and_create_by_columns.yaml

```diff
@@ -0,0 +1,67 @@
+databaseChangeLog:
+- changeSet:
+    id: Add created_date and created_by columns
+    author: yuhuyoyo
+    changes:
+    - addColumn:
+        tableName: workspace
+        columns:
+        - column:
+            name: created_date
+            type: timestamptz
+            constraints:
+               nullable: false
+            remarks: |
+               Timestamp when this workspace is created
+            defaultValueComputed: now()
```

Comment:
Should we validate anywhere that activity log matches these two columns? (A downside of adding these columns, is that you might have bugs where the two places get out of sync.)

Even if we don't validate, on further thought, I think it would be nice if the date here matched the activity log, rather than be milliseconds or nanoseconds off. I don't think that can happen with defaultValueComputed...

Reply:
I assume this is to set the value for existing rows. Otherwise, they would be null and violate the constraint. The code does not use the `now()` I don't think. For those rows, the activity log will not match the row.

A more complex alternative would be to add one more schema change after the SQL to update the column removing the default value. Not sure it is worth it, but could be done. wdyt

Reply:
@yuhuyoyo correct me if I'm wrong. I think line 59 of this file is for filling in existing rows.

In WorkspaceDao.createWorkspace, the SQL sets created_by_email and not created_date. I think because created_date is set by `defaultValueComputed: now()`. So I think the two places will be out of sync.

A much simpler alternative to this PR is to just add change_date to the index in workspace_activity_log.

Reply:
yeah line 59 is for backfilling for workspace.

we currently don't log for resource and folder creation so there's no way to backfill those.

Reply:
When we implement locking, there will be an update to the object row at the end of the flight. That will make the times closer, but not identical. I think we'll need help from Stairway if we want them to be identical.

Reply:
i think the miliseconds differences are tolerable. UI doesn't even care that level of granularity https://screenshot.googleplex.com/5LVr8zZ8RKdDsME. 

i added comment in the database as Melissa suggested.